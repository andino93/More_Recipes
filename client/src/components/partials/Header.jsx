import React from 'react';
import ReactDOM from 'react-dom';

/**
 * @export
 * @class NavbarComponent
 * @extends {React.Component}
 */
export default class HeaderComponent extends React.Component {
    render() {
        return (
            <nav className="navbar navbar-expand-lg navbar-light bg-navbar fixed-top">
                <a className="navbar-brand" href="index.html">MoreRecipe</a>
                <button
                    className="navbar-toggler"
                    type="button"
                    data-toggle="collapse"
                    data-target="#navbarSupportedContent"
                    aria-controls="navbarSupportedContent"
                    aria-expanded="false"
                    aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>

                <div
                    className="collapse navbar-collapse justify-content-end"
                    id="navbarSupportedContent">
                    <div className="navbar-nav ml-auto btn-group">
                        <button
                            type="button"
                            className="btn btn-outline-secondary"
                            data-toggle="modal"
                            data-target="#exampleModalLong">
                            <i className="fa fa-user-circle"></i>
                            My Account
                        </button>
                    </div>
                </div>
            </nav>
        );
    }
}
