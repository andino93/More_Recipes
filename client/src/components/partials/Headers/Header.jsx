import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'react-proptypes';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { logoutAction } from '../../../actions/auth/signinAction';

/**
 * @export
 * 
 * @class HeaderComponent
 * 
 * @extends {React.Component}
 */
class Header extends Component {
  /**
 * Creates an instance of Header.
 * 
 * @param {any} props 
 * 
 * @memberof Header
 */
  constructor(props) {
    super(props);
    this.logoutUser = this.logoutUser.bind(this);
  }
  /**
     *
     * @returns {XML} XML/JSX
     * 
     * @memberof HeaderComponent
     */
  logoutUser() {
    this.props.logoutAction();
  }
  /**
   *
   *
   * @returns {XML} XML/JSX
   * 
   * @memberof HeaderComponent
   */
  render() {
    return (
      <div>
        { (window.localStorage.token) ?
          <nav 
            className="navbar navbar-expand-sm
          navbar-light bg-navbar
          fixed-top">
            <Link to="/" className="navbar-brand" id="home">
              More Recipes
            </Link>
            <button className="navbar-toggler"
              type="button" data-toggle="collapse"
              data-target="#navbarSupportedContent"
              aria-controls="navbarSupportedContent"
              aria-expanded="false"
              aria-label="Toggle navigation">
              <span className="navbar-toggler-icon" />
            </button>

            <div className="collapse navbar-collapse"
              id="navbarSupportedContent">
              <ul className="navbar-nav ml-auto">
                <li className="nav-item dropdown">
                  <Link className="nav-link dropdown-toggle"
                    to="#" id="navbarDropdown"
                    role="button"
                    data-toggle="dropdown"
                    aria-haspopup="true"
                    aria-expanded="false">
                  My Account
                  </Link>
                  <div className="dropdown-menu dropdown-menu-right mr-20"
                    aria-labelledby="navbarDropdown">
                    <Link to="/profile" className="dropdown-item">
                      <i className="fa fa-user-circle"/> My Profile
                    </Link>
                    <Link to="/user-recipes" className="dropdown-item">
                      <i className="fa fa-list"/> My Recipes
                    </Link>
                    <Link to="/my-favorite" className="dropdown-item">
                      <i className="fa fa-star" /> My Favorites
                    </Link>
                    <Link to="/addRecipe" className="dropdown-item">
                      <i className="fa fa-plus" /> Add Recipes
                    </Link>
                    <div className="dropdown-divider" />
                    <Link to="#"
                      className="dropdown-item"
                      onClick={this.logoutUser}>
                      <i className="fa fa-sign-out" />
                    Log out
                    </Link>
                  </div>
                </li>
              </ul>
            </div>
          </nav>
          :
          <nav 
            className="navbar
            navbar-expand-lg navbar-light bg-navbar fixed-top">
            <Link to="/" className="navbar-brand" >MoreRecipe</Link>
            <button
              className="navbar-toggler"
              type="button"
              data-toggle="collapse"
              data-target="#navbarSupportedContent"
              aria-controls="navbarSupportedContent"
              aria-expanded="false"
              aria-label="Toggle navigation">
              <span className="navbar-toggler-icon" />
            </button>
            <div
              className="collapse navbar-collapse justify-content-end"
              id="navbarSupportedContent">
              <div className="navbar-nav ml-auto btn-group">
                <button
                  type="button"
                  className="btn btn-outline-secondary"
                  data-toggle="modal"
                  data-target="#signInModal"
                  id="signin">
                Sign In
                  <i className="fa fa-sign-in" />
                </button>&nbsp;&nbsp;
                <button
                  type="button"
                  className="btn btn-outline-secondary"
                  data-toggle="modal"
                  data-target="#signUpModal"
                  id="signup">
                Sign Up
                  <i className="fa fa-user-plus" />
                </button>
              </div>
            </div>
          </nav>
        }
      </div>
    );
  }
}

Header.propTypes = {
  logoutAction: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  signInState: state.signinReducer
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({ logoutAction }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Header);