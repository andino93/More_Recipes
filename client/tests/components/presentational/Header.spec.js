import React from 'react';
import expect from 'expect';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { configure } from 'enzyme';
import sinon from 'sinon';
import mockStore from '../../__mocks__/mockStore';
import history from '../../../src/utils/history';
import localStorageMock from '../../__mocks__/localStorageMock';
import mockData from "../../__mocks__/mockData";
import mountWrap from '../../__mocks__/utils';
import Header from '../../../src/components/partials/Headers/Header.jsx';

// noinspection JSAnnotator
window.localStorage = localStorageMock;
window.localStorage.token = mockData.token;

configure({ adapter: new Adapter() });

describe('<Header>', () => {
  const store = mockStore({});
  const connectedHeader = mountWrap(Header, store, {});
  it('should render the Header component', (done) => {
    sinon.stub(history, 'push');
    expect(connectedHeader.find('button').at(0).prop('type')).toEqual('button');
    expect(toJson(connectedHeader)).toMatchSnapshot();
    connectedHeader.find('Header').find('Link').at(6).simulate('click');
    done();
  });
});
