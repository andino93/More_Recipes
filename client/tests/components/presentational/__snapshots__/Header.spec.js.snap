// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> should render the Header component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Header)>
        <Header
          logoutAction={[Function]}
        >
          <div>
            <nav
              className="navbar navbar-expand-sm navbar-light bg-navbar fixed-top"
            >
              <Link
                className="navbar-brand"
                id="home"
                replace={false}
                to="/"
              >
                <a
                  className="navbar-brand"
                  href="/"
                  id="home"
                  onClick={[Function]}
                >
                  More Recipes
                </a>
              </Link>
              <button
                aria-controls="navbarSupportedContent"
                aria-expanded="false"
                aria-label="Toggle navigation"
                className="navbar-toggler"
                data-target="#navbarSupportedContent"
                data-toggle="collapse"
                type="button"
              >
                <span
                  className="navbar-toggler-icon"
                />
              </button>
              <div
                className="collapse navbar-collapse"
                id="navbarSupportedContent"
              >
                <ul
                  className="navbar-nav ml-auto"
                >
                  <li
                    className="nav-item dropdown"
                  >
                    <Link
                      aria-expanded="false"
                      aria-haspopup="true"
                      className="nav-link dropdown-toggle"
                      data-toggle="dropdown"
                      id="navbarDropdown"
                      replace={false}
                      role="button"
                      to="#"
                    >
                      <a
                        aria-expanded="false"
                        aria-haspopup="true"
                        className="nav-link dropdown-toggle"
                        data-toggle="dropdown"
                        href="#"
                        id="navbarDropdown"
                        onClick={[Function]}
                        role="button"
                      >
                        My Account
                      </a>
                    </Link>
                    <div
                      aria-labelledby="navbarDropdown"
                      className="dropdown-menu dropdown-menu-right mr-20"
                    >
                      <Link
                        className="dropdown-item"
                        replace={false}
                        to="/profile"
                      >
                        <a
                          className="dropdown-item"
                          href="/profile"
                          onClick={[Function]}
                        >
                          <i
                            className="fa fa-user-circle"
                          />
                           My Profile
                        </a>
                      </Link>
                      <Link
                        className="dropdown-item"
                        replace={false}
                        to="/user-recipes"
                      >
                        <a
                          className="dropdown-item"
                          href="/user-recipes"
                          onClick={[Function]}
                        >
                          <i
                            className="fa fa-list"
                          />
                           My Recipes
                        </a>
                      </Link>
                      <Link
                        className="dropdown-item"
                        replace={false}
                        to="/my-favorite"
                      >
                        <a
                          className="dropdown-item"
                          href="/my-favorite"
                          onClick={[Function]}
                        >
                          <i
                            className="fa fa-star"
                          />
                           My Favorites
                        </a>
                      </Link>
                      <Link
                        className="dropdown-item"
                        replace={false}
                        to="/addRecipe"
                      >
                        <a
                          className="dropdown-item"
                          href="/addRecipe"
                          onClick={[Function]}
                        >
                          <i
                            className="fa fa-plus"
                          />
                           Add Recipes
                        </a>
                      </Link>
                      <div
                        className="dropdown-divider"
                      />
                      <Link
                        className="dropdown-item"
                        onClick={[Function]}
                        replace={false}
                        to="#"
                      >
                        <a
                          className="dropdown-item"
                          href="#"
                          onClick={[Function]}
                        >
                          <i
                            className="fa fa-sign-out"
                          />
                          Log out
                        </a>
                      </Link>
                    </div>
                  </li>
                </ul>
              </div>
            </nav>
          </div>
        </Header>
      </Connect(Header)>
    </Router>
  </BrowserRouter>
</Provider>
`;
